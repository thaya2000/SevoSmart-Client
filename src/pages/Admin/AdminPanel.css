/* AdminPanel.css */


.admin-panelflex {
    display: flex;
    flex-direction: row;
  }
  
  .admin-sidebar {
    background-color: #2d3748;
    width: 16rem;
    padding: 1rem;
    color: white;
  }
  
  .admin-table {
    flex: 1;
    padding: 1rem;
    overflow-y: auto;
  }
  
  .table-auto {
    width: 100%;
  }
  
  .table-auto th,
  .table-auto td {
    padding: 0.5rem;
    border: 1px solid #e2e8f0;
  }
  
  .table-auto thead th {
    background-color: #edf2f7;
    color: #2d3748;
  }
  
  .table-auto tbody tr:nth-child(even) {
    background-color: #f7fafc;
  }
  
  .fixed {
    position: fixed;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
  }
  
  .inset-0 {
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
  }
  
  .z-50 {
    z-index: 50;
  }
  
  .bg-black {
    background-color: black;
  }
  
  .bg-opacity-40 {
    background-color: rgba(0, 0, 0, 0.4);
  }
  
  .w-16 {
    width: 4rem;
  }
  
  .h-16 {
    height: 4rem;
  }
  
  .border-4 {
    border-width: 4px;
  }
  
  .border-blue-500 {
    border-color: #3b82f6;
  }
  
  .border-t-transparent {
    border-top-color: transparent;
  }
  
  .rounded-full {
    border-radius: 9999px;
  }
  
  .animate-spin {
    animation: spin 1s linear infinite;
  }
  
  @keyframes spin {
    from {
      transform: rotate(0deg);
    }
    to {
      transform: rotate(360deg);
    }
  }
  
  
  @media (max-width: 768px) {
    .admin-panelflex {
      flex-direction: column;
    }
  
    .admin-sidebar {
      width: 100%;
      padding: 0.5rem;
      font-size: 0.8rem; 
    }
  
    .admin-sidebar h2 {
      font-size: 1.25rem; 
    }
  
    .admin-sidebar ul li a {
      font-size: 0.8rem; 
    }
    .admin-table {
      padding: 0.5rem;
      font-size: 0.8rem; 
    }
   
    .admin-tableh {
      font-size: 1rem; 
      margin-top: 1rem;
      font-weight: 700;
    }
  
    .admin-table th,
    .admin-table td {
      font-size: 0.8rem; 
    }
  }
