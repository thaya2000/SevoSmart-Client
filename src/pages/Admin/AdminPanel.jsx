import React, { useState, useEffect } from "react";
import { Link } from "react-router-dom";
import axios from "axios";
import RambousLoader from "../../routes/RambousLoader";
import "./AdminPanel.css";

const AdminPanel = () => {
  const [loading, setLoading] = useState(false);
  const [orders, setOrders] = useState([]);
  const [serialNumbers, setSerialNumbers] = useState({});

  useEffect(() => {
    fetchOrders();
  }, []);

  const fetchOrders = async () => {
    setLoading(true);
    try {
      const response = await axios.get("/api/v1/user/allOrders");
      setOrders(response.data); // Assuming your API response is an array of order objects
      generateSerialNumbers(response.data);
      console.log(response.data);
    } catch (error) {
      console.error("Error fetching orders:", error);
    } finally {
      setLoading(false);
    }
  };

  const generateSerialNumbers = (orders) => {
    const serials = {};
    orders.forEach((order, index) => {
      serials[order.orderNumber] = index + 1;
    });
    setSerialNumbers(serials);
  };

  return (
    <div>
      {loading ? (
        <RambousLoader />
      ) : (  
        <div className="admin-panelflex h-screen">
          <div className="admin-sidebar bg-gray-800 w-100% p-4 text-white">
            <h2 className="text-2xl font-bold mb-4">Admin Panel</h2>
            <ul className="space-y-4">
              <li>
                <Link to="/users">Users</Link>
              </li>
              <li>
                <Link to="/admin/products">Accessories</Link>
              </li>
              <li>
                <Link to="/past-projects">Past Projects</Link>
              </li>
              <li>
                <Link to="/news-admin">News</Link>
              </li>
              <li>
                <Link to="/logout">Logout</Link>
              </li>
            </ul>
          </div>

          <div className="admin-table flex-1 p-4 overflow-y-auto">
            <h1 className="admin-tableh text-2xl font-bold mb-4">
              Order Details
            </h1>
            <div className="overflow-x-auto">
              <table className="table-auto w-full">
                <thead>
                  <tr className="bg-gray-200 text-gray-800">
                    <th className="px-4 py-2">No</th>
                    <th className="px-4 py-2">Customer Name</th>
                    <th className="px-4 py-2">Order date</th>
                    <th className="px-4 py-2">Phone No</th>
                    <th className="px-4 py-2">Status</th>
                    <th className="px-4 py-2">Order Amount</th>
                    <th className="px-4 py-2">Address</th>
                    <th className="px-4 py-2">Actions</th>
                  </tr>
                </thead>
                <tbody>
                  {orders.map((order) => (
                    <tr key={order.orderNumber}>
                      <td className="border px-4 py-2">
                        {serialNumbers[order.orderNumber]}
                      </td>
                      <td className="border px-4 py-2">{order.customerName}</td>
                      <td className="border px-4 py-2">{order.orderDate}</td>
                      <td className="border px-4 py-2">
                        {order.customerPhoneNo}
                      </td>
                      <td className="border px-4 py-2">{order.orderStatus}</td>
                      <td className="border px-4 py-2">{order.orderAmount}</td>
                      <td className="border px-4 py-2">
                        {order.orderBillingAddress}
                      </td>
                      <td className="border px-4 py-2">
                        <div className="flex flex-row justify-center">
                          <Link
                            to={`/admin/order-details/${order.orderNumber}`}
                            className="bg-green-500 hover:bg-green-700 text-white font-bold py-1 px-4 mr-2 rounded"
                          >
                            View
                          </Link>
                        </div>
                      </td>
                    </tr>
                  ))}
                </tbody>
              </table>
            </div>
          </div>
        </div>
      )}
    </div>
  );
};

export default AdminPanel;
