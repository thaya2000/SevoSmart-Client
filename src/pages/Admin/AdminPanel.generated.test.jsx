import { describe, it, expect, vi } from 'vitest';
import { render } from '@testing-library/react';
import AdminPanel from './AdminPanel';

// Mocking modules
vi.mock('react-router-dom', () => ({
  Link: vi.fn(() => <div>Link</div>) // Stubbing the Link component
}));
vi.mock('axios');

describe('<AdminPanel>', () => {
  it('should render component', () => {
    const { asFragment } = render(<AdminPanel />);
    expect(asFragment()).toMatchSnapshot();
  });
});
