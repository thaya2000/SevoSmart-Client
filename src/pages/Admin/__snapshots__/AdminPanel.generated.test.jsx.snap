// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<AdminPanel> > should render component 1`] = `
<DocumentFragment>
  <div
    class="flex h-screen"
  >
    <div
      class="bg-gray-800 w-64 p-4 text-white"
    >
      <h2
        class="text-2xl font-bold mb-4"
      >
        Admin Panel
      </h2>
      <ul
        class="space-y-4"
      >
        <li>
          <div>
            Link
          </div>
        </li>
        <li>
          <div>
            Link
          </div>
        </li>
        <li>
          <div>
            Link
          </div>
        </li>
        <li>
          <div>
            Link
          </div>
        </li>
        <li>
          <div>
            Link
          </div>
        </li>
      </ul>
    </div>
    <div
      class="flex-1 p-4 overflow-y-auto"
    >
      <h2
        class="text-2xl font-bold mb-4"
      >
        Order Details
      </h2>
      <div
        class="overflow-x-auto"
      >
        <table
          class="table-auto w-full"
        >
          <thead>
            <tr
              class="bg-gray-200 text-gray-800"
            >
              <th
                class="px-4 py-2"
              >
                No
              </th>
              <th
                class="px-4 py-2"
              >
                Customer Name
              </th>
              <th
                class="px-4 py-2"
              >
                Phone
              </th>
              <th
                class="px-4 py-2"
              >
                Email
              </th>
              <th
                class="px-4 py-2"
              >
                Address
              </th>
              <th
                class="px-4 py-2"
              >
                Product Name
              </th>
              <th
                class="px-4 py-2"
              >
                Quantity
              </th>
              <th
                class="px-4 py-2"
              >
                Status
              </th>
              <th
                class="px-4 py-2"
              >
                Actions
              </th>
            </tr>
          </thead>
          <tbody />
        </table>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
